src/EPR.Calculator.API/Controllers/CalculatorInternalController.cs

```diff
@@ -0,0 +1,137 @@
+﻿using EPR.Calculator.API.Data;
+using EPR.Calculator.API.Data.DataModels;
+using EPR.Calculator.API.Dtos;
+using EPR.Calculator.API.Models;
+using EPR.Calculator.API.Validators;
+using EPR.Calculator.API.Wrapper;
+using Microsoft.AspNetCore.Mvc;
+
+namespace EPR.Calculator.API.Controllers
+{
+    [Route("v1/internal")]
+    [ApiController]
+    public class CalculatorInternalController : ControllerBase
+    {
+        private readonly ApplicationDBContext context;
+        private readonly IRpdStatusDataValidator rpdStatusDataValidator;
+        private readonly IOrgAndPomWrapper wrapper;
+
+        public CalculatorInternalController(ApplicationDBContext context,
+                                            IRpdStatusDataValidator rpdStatusDataValidator,
+                                            IOrgAndPomWrapper wrapper)
+        {
+            this.context = context;
+            this.rpdStatusDataValidator = rpdStatusDataValidator;
+            this.wrapper = wrapper;
+        }
+
+        [HttpPost]
+        [Route("rpdStatus")]
+        public IActionResult UpdateRpdStatus([FromBody] UpdateRpdStatus request)
+        {
+            var runId = request.RunId;
+            var calcRun = this.context.CalculatorRuns.SingleOrDefault(run => run.Id == runId);
+            var runClassifications = this.context.CalculatorRunClassifications.ToList();
+
+            var validationResult = this.rpdStatusDataValidator.IsValidRun(calcRun, runId, runClassifications);
+            if (!validationResult.isValid)
+            {
+                return StatusCode(validationResult.StatusCode, validationResult.ErrorMessage);
+            }
+
+            if (!request.isSuccessful && calcRun != null)
+            {
+                calcRun.CalculatorRunClassificationId = runClassifications.Single(x => x.Status == RunClassiciations.ERROR.ToString()).Id;
+                this.context.SaveChanges();
+                return new ObjectResult(null) { StatusCode = StatusCodes.Status201Created };
+            }
+
+            var vr = this.rpdStatusDataValidator.IsValidSuccessfulRun(runId);
+            if (!vr.isValid)
+            {
+                return StatusCode(vr.StatusCode, vr.ErrorMessage);
+            }
+
+            using (var transaction = this.context.Database.BeginTransaction())
+            {
+                try
+                {
+                    var stagingOrganisationData = this.wrapper.GetOrganisationData();
+                    foreach (var organisation in stagingOrganisationData)
+                    {
+                        var calcOrganisationMaster = new CalculatorRunOrganisationDataMaster
+                        {
+                            CalendarYear = "2023", //Take the financial year from Calc Run table and Derive the Calendar year
```

Comment:
CalendarYear is here 2023 and below in CalculatorRunPomDataMaster is 2024-25 both looks different in format and year. Is it correct?